addons/sourcemod/scripting/shavit-tas.sp

```diff
@@ -0,0 +1,1395 @@
+	Shavit_PrintToChat(client, "+autostrafer - When bound hold to use wigglehack");
+	Shavit_PrintToChat(client, "!tasmenu - Toggles TAS Menu");
+	return Plugin_Handled;
+}
+
+public Action CommandListener_TAS(int client, const char[] command, int args)
+{
+	gB_TASMenu[client] = true;
+	DrawPanel(client);
+	return Plugin_Continue;
+}
+
+public Action CommandListener_JoinTeam(int client, const char[] command, int args)
+{
+	if(gB_TAS[client])
+	{
+		gI_Status[client] = RUN;
+		gF_TASTime[client] = 0.0;
+		gI_IndexCounter[client] = 0;
+		ClearArray(gA_Frames[client]);
+		FakeClientCommandEx(client, "sm_r");
+	}
+	return Plugin_Continue;
+}
+
+public Action CommandListener_PlusRewind(int client, const char[] command, int args)
+{
+	if(!gB_TAS[client])
+	{
+		return Plugin_Continue;
+	}
+	gI_Status[client] = BACKWARD;
+	return Plugin_Handled;
+}
+
+public Action CommandListener_PlusForward(int client, const char[] command, int args)
+{
+	if(!gB_TAS[client])
+	{
+		return Plugin_Continue;
+	}
+	gI_Status[client] = FORWARD;
+	return Plugin_Handled;
+}
+
+public Action CommandListener_MinusRewindOrForward(int client, const char[] command, int args)
+{
+	if(!gB_TAS[client])
+	{
+		return Plugin_Continue;
+	}
+	gI_Status[client] = PAUSED;
+	return Plugin_Handled;
+}
+
+public void OnClientPutInServer(int client)
+{
+	
+	if(gA_Frames[client] != INVALID_HANDLE)
+	{
+		ClearArray(gA_Frames[client]);
+	}
+	else
+	{
+		gA_Frames[client] = CreateArray(11, 0);
+	}
+
+	gF_CounterSpeed[client] = 1.0;
+	gF_TASTime[client] = 0.0;
+	gF_Timescale[client] = 1.0;
+	gI_Status[client] = RUN;
+	gB_TASMenu[client] = true;
+	gB_AutoStrafeEnabled[client] = false;
+	gB_SilentStrafe[client] = false;
+	gI_Type[client] = Type_SurfOverride;
+	gF_Power[client] = 1.0;
+}
+
+public Action OnPlayerRunCmd(int client, int &buttons, int &impulse, float vel[3], float angles[3], int &weapon, int &subtype, int &cmdnum, int &tickcount, int &seed, int mouse[2])
+{
+	if (IsValidClient(client, true) && gB_TAS[client])
+	{
+
+		if(!gB_TAS[client])
+		{
+			return Plugin_Continue;
+		}
+
+		DrawPanel(client);
+
+		if(Shavit_GetTimerStatus(client) != Timer_Running)
+		{
+			return Plugin_Continue;
+		}
+		else if(IsPlayerAlive(client) && !IsFakeClient(client))
+		{
+			if(gI_Status[client] == RUN)
+			{ // Record Frames
+				//SetEntPropFloat(client, Prop_Send, "m_flLaggedMovementValue", gF_Timescale[client]);
+				float fTimeScale = GetEntPropFloat(client, Prop_Data, "m_flLaggedMovementValue");
+				gF_TimescaleTicksPassed[client] += fTimeScale;
+
+				float fDifference = angles[1] - gF_LastAngle[client];
+				if (fDifference > 180.0)
+				{
+					fDifference -= 360.0;
+				}
+				else if(fDifference < -180.0)
+				{
+					fDifference += 360.0;
+				}
+				/*
+							AUTO STRAFER START
+														*/
+				if(buttons & IN_FORWARD && vel[0] <= 50.0)
+				{
+					vel[0] = 450.0;
+				}
+
+				float yaw_change = 0.0;
+				if(vel[0] > 50.0)
+				{
+					yaw_change = 30.0 * FloatAbs(30.0 / vel[0]);
```

Comment:
shouldn't this be that wish speed variable, instead of a hard coded 30? and no snakes